src/AtmosphereRenderer.cpp

```diff
@@ -75,14 +76,19 @@ void AtmosphereRenderer::setWorldTransform(glm::dmat4 const& transform) {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void AtmosphereRenderer::setCloudTexture(
-    std::shared_ptr<VistaTexture> const& texture, float height) {
-  if (mCloudTexture != texture || mCloudHeight != height) {
-    mCloudTexture = texture;
-    mCloudHeight  = height;
-    mShaderDirty  = true;
-    mUseClouds    = texture != nullptr;
+void AtmosphereRenderer::setClouds(std::string const& textureFile, float height) {
+
+  if (mCloudTextureFile != textureFile) {
+    mCloudTextureFile = textureFile;
+    mCloudTexture.reset();
+    if (textureFile != "") {
```

Comment:
```suggestion
    if (!textureFile.empty()) {
```

Reply:
You're right but in order to avoid too many conflicts with the `feature/clang-tidy` I will do this after it got merged.